pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -60,6 +60,59 @@ def import_params_torch_to_returnn(self, *, layer: LinearLayer, torch_module: Li
       layer.params["b"].load(torch_module.bias.detach().numpy(), session=session)
 
 
+class FunctionalLinear(Module):
+  is_original_torch_module = False
+
+  def create_returnn_layer_dict(self, input: Tensor, weight: Tensor, bias: Optional[Tensor] = None) -> Dict[str, Any]:
+    assert len(weight.shape) == 2
+    out_features, in_features = weight.shape
+    if bias is not None:
+      assert bias.shape == (out_features,)
+    return {
+      "class": "linear", "from": self._get_input_layer_name(input),
+      "n_out": out_features,
+      "with_bias": bias is not None,
+      "bias": self._get_input_layer_name(bias) if bias is not None else None,
+      "weights": self._get_input_layer_name(weight),
+      "use_transposed_weights": True,
+      "activation": None}
+
+
+class Matmul(Module):
+  """
+  Maps to RETURNN DotLayer
+  """
+  is_original_torch_module = False
+
+  def create_returnn_layer_dict(self, *inputs: Tensor, **kwargs) -> Dict[str, Any]:
+    sources = [self._get_input_layer_name(source) for source in inputs]
+    assert len(sources) == 2
+
+    assert len(inputs[0].shape) >= 2, "not implemented otherwise"
+    assert len(inputs[1].shape) >= 2, "not implemented otherwise"
+    red1 = inputs[0].returnn_naming_entry.get_returnn_axis_description(-1)
+    red2 = inputs[1].returnn_naming_entry.get_returnn_axis_description(-2)
+    var1 = [inputs[0].returnn_naming_entry.get_returnn_axis_description(-2)]
+    var2 = [inputs[1].returnn_naming_entry.get_returnn_axis_description(-1)]
```

Comment:
See other layers. Make this consistent. You can use `self._get_axis...` (or sth like that). Do not access `returnn_naming_entry`.

Reply:
`self._get_input_axis_to_returnn`